test/e2e/aws_eip_for_nlb_test.go

```diff
@@ -0,0 +1,720 @@
+	kubeClient, err := createClient()
+	if err != nil {
+		t.Fatalf("Test Skipped %s", err)
+	}
+	decodedAccessKeyID, decodedSecretAccessKey, err := getIDAndKey(kubeClient)
+	t.Logf("decodedAccessKeyID: %v, decodedSecretAccessKey: %v", decodedAccessKeyID, decodedSecretAccessKey)
+	if err != nil {
+		t.Fatalf("Test Skipped %s", err)
+	}
+	region, err := getRegion()
+	if err != nil {
+		t.Fatalf("Test Skipped %s", err)
+	}
+	// Set up an AWS session
+	sess, err := createSession(region, decodedAccessKeyID, decodedSecretAccessKey)
+	if err != nil {
+		t.Fatalf("Test Skipped %s", err)
+	}
+	// Create an EC2 service client
+	svc := ec2.New(sess)
+
+	clusterName, err := getClusterName()
+	if err != nil {
+		t.Fatalf("test skipped: %v", err)
+	}
+
+	vpcID, err := getVPCID(clusterName, svc)
+	if err != nil {
+		t.Fatalf("Test Skipped %s", err)
+	}
+
+	// Describe addresses to get unassociated EIPs
+	describeAddressesInput := &ec2.DescribeAddressesInput{}
+	describeAddressesOutput, err := svc.DescribeAddresses(describeAddressesInput)
+	if err != nil {
+		t.Fatalf("failed to describe addresses, %v", err)
+	}
+
+	var unassociatedEIPs []struct {
+		PublicIp     string
+		AllocationId string
+	}
+
+	for _, address := range describeAddressesOutput.Addresses {
+		if address.AssociationId == nil {
+			// Check if the EIP is associated with a network interface in the specific VPC
+			if address.NetworkInterfaceId != nil {
+				describeNetworkInterfacesInput := &ec2.DescribeNetworkInterfacesInput{
+					NetworkInterfaceIds: []*string{address.NetworkInterfaceId},
+				}
+				describeNetworkInterfacesOutput, err := svc.DescribeNetworkInterfaces(describeNetworkInterfacesInput)
+				if err != nil {
+					t.Fatalf("failed to describe network interfaces, %v", err)
+				}
+
+				for _, networkInterface := range describeNetworkInterfacesOutput.NetworkInterfaces {
+					if networkInterface.VpcId != nil && *networkInterface.VpcId == vpcID {
+						unassociatedEIPs = append(unassociatedEIPs, struct {
+							PublicIp     string
+							AllocationId string
+						}{
+							PublicIp:     aws.StringValue(address.PublicIp),
+							AllocationId: aws.StringValue(address.AllocationId),
+						})
+					}
+				}
+			} else {
+				// If there's no associated network interface, assume it's an unassociated EIP
+				unassociatedEIPs = append(unassociatedEIPs, struct {
+					PublicIp     string
+					AllocationId string
+				}{
+					PublicIp:     aws.StringValue(address.PublicIp),
+					AllocationId: aws.StringValue(address.AllocationId),
+				})
+			}
+		}
+	}
+
+	if len(unassociatedEIPs) == 0 {
+		t.Log("No unassociated EIPs found.")
+	}
+
+	// Release each unassociated EIP
+	for _, eip := range unassociatedEIPs {
+		releaseAddressInput := &ec2.ReleaseAddressInput{
+			AllocationId: aws.String(eip.AllocationId),
+		}
+		_, err := svc.ReleaseAddress(releaseAddressInput)
+		if err != nil {
+			t.Logf("Failed to release %s EIP with Allocation ID: %s\n", eip.PublicIp, eip.AllocationId)
```

Comment:
This should probably be a error so it fails the test if it fails to cleanup, so we are alerted to a cleanup issue. 

```suggestion
			t.Errorf("Failed to release %s EIP with Allocation ID: %s\n", eip.PublicIp, eip.AllocationId)
```

Reply:
done